### ISSUE TITLE
[web] Default bootstrapping logic should not wait for `load` event

### ISSUE THREAD

**Issue by eyebrowsoffire** [flutter/flutter#133805]
The default code that we insert in index.html when bootstrapping a flutter app is as follows:
```
    window.addEventListener('load', function (ev) {
      // Download main.dart.js
      _flutter.loader.loadEntrypoint({
        onEntrypointLoaded: function (engineInitializer) {
          engineInitializer.initializeEngine().then(function (appRunner) {
            appRunner.runApp();
          });
        }
      });
    });
```
This, however, is not good because it means that actually initializing the engine is blocked on *all* asynchronous scripts and resources on the page. As an example, I loaded the skwasm Material 3 demo here: https://flutterweb-wasm--skwasm-1s3iqgbs.web.app/

Taking a perf profile I saw the following waterfall:
![image](https://github.com/flutter/flutter/assets/26911996/0830e680-387b-4c2c-b6b9-1049f44e187d)
As can be seen from the trace, the highlighted "Google Tag Manager" request (which is just analytics that could otherwise be completely asynchronous) is blocking the window load event, and engine initialization doesn't start until that is completely resolved. I went ahead and did a devtools local override of the code in `index.html` to listen to the `DOMContentLoaded` event instead, and the waterfall is much healthier:

![image](https://github.com/flutter/flutter/assets/26911996/2ffb03e0-6694-43a4-8246-9d510d3ce478)

I think instead of waiting on `DOMContentLoaded`, the only thing we actually care about here specifically is that the `flutter.js` script is loaded, so I think we should just explicitly wait on that event rather than tying it to a browser load event.

**Comment by dvoloshyn** [comment]
Agree with the proposal.
I have a loading indicator with an `<img>` and with default flutter template (`load` event) it does not start fetching the engine until the image is loaded.
Docs clearly say that `DOMContentLoaded` will not fire until all `defer` scripts are loaded and evaluated : [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#defer).
So, I changed to `DOMContentLoaded` and now it does not wait for image to load.
Also, using `fetchpriority="high"` should ensure that `flutter.js` is requested first if there are other external scripts: [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#fetchpriority](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#fetchpriority)

**Comment by reke592** [comment]
Hi, may we request to elevate the priority of this issue back to P1. It's crucial for web deployments because the deferred import functionality appears to be ineffective as long as we encounter this initial loading problem.

In our case, we have a huge main.dar.js file, which is 2MB when gzipped. To mitigate this, we've created a separate bootstrap dart file for deferred loading. Our expectation was for these files to load in parallel. However, they are loading sequentially. While this does split the size, another issue arises from the loading of certain parts being blocked by other requests.

![image](https://github.com/flutter/flutter/assets/17094894/f6c95f1f-d450-499a-9d8e-ec8df97b3f47)
![image](https://github.com/flutter/flutter/assets/17094894/b40b4bf6-1aeb-4f62-95e7-46d20effd383)

**Comment by eyebrowsoffire** [comment]
FWIW, you don't actually have to wait for the 'load' event on the document before running flutter.js stuff or loading the app, that's just the default behavior when doing `flutter create`. That's specifically what this issue tracks, that we're essentially recommending a workflow that is slower than necessary. You can, if you want, write your own bootstrapping script that just waits on the `flutter.js` script or the DOMContentLoaded event instead.

I'd like to avoid making this a P1 for specific reasons relating to the web engine team's triage process, but rest assured that some changes are coming down the pipe very soon that should improve things in several ways, including (1) generally faster and more parallelized waterfall on startup, (2) simpler startup invocation in general, and (3) better default behavior on `flutter create`. They will probably obviate most of the issues here. I'll report back on this issue when these improvements land.

**Comment by eyebrowsoffire** [comment]
The bootstrapper no longer waits for the load event.
